module-services/service-desktop/ServiceDesktop.cpp

```diff
@@ -53,6 +54,19 @@ auto ServiceDesktop::getSerialNumber() const -> std::string
     return settings->getValue(std::string("factory_data/serial"), settings::SettingsScope::Global);
 }
 
+auto ServiceDesktop::getDeviceToken() const -> std::string
+{
+    std::string tokenSeed = getDeviceUniqueId();
+    SHA256 sha256;
+
+    return sha256(tokenSeed);
+}
+
+auto ServiceDesktop::getDeviceUniqueId() const -> std::string
+{
+    return settings->getValue(std::string("gs_device_unique_id"), settings::SettingsScope::Global);
```

Comment:
Not sure if ServiceDesktop should take the value of this setting directly via settings API.
Probably it should use API provided by ServiceAppMgr (owner of the setting) to obtain setting value.
Please double check with https://appnroll.atlassian.net/wiki/spaces/MFP/pages/1281917632/Settings+API+usage+guideline+followup#Usage-guideline

Reply:
Yes, but I think this is for the owner of a specified setting.
In _Don'ts_ we have:
_use other applications/services settings - use API to get values_
TBH I'm not sure which way is correct.
Creating and using additional API in `ServiceAppMgr` seems like adding code bloat if we can access it directly.
On the other hand, by using `getValue()` on the value that is not owned by `ServiceDesktop` we are simply taking advantage of our poor system architecture. The truth is that this setting probably shouldn't be accessible in the global scope.

@pholat Could you advise here?
Can we use `settings->getValue` in `ServiceDesktop` for getting setting value owned by `ServiceAppMgr`?

Reply:
tbh this discussion was done multiple times. This was partially meant to be solved by a proposition by Wiktor to create a static API for immediate access data. 
This way we could ask owners for their (owned) data. Not use global data everywhere.

But - for UID isn't there already API to get it? AFAIR UID data was loaded from separate memory to the settings then there was some wrapper to get it.

Reply:
This UID is different from Serial Number, it changes after factory reset and is used as a secret substrate for MC to encrypt user data cache.

Reply:
ok, so why exactly is this UID global? not applocal? and did you check the devel.sql at the top to be working with what's added here? (i.e. with presenting it with some data, for example "2" or something?)

Reply:
Changed the UID to be AppLocal as suggested.